psh.proto

```diff
@@ -1,12 +1,25 @@
 syntax ="proto3";
 
-import "google/protobuf/empty.proto";
 package psh;
 
+import "google/protobuf/empty.proto";
+
 service PshService {
-    rpc Heartbeat(google.protobuf.Empty) returns (PshResponse);
+  rpc Heartbeat(google.protobuf.Empty) returns (PshResponse);
+  rpc SendHostInfo(HostInfoRequest) returns (HostInfoResponse);
 }
 
 message PshResponse {
-    string resp = 1;
-}
\ No newline at end of file
+  string resp = 1;
+}
+
+message HostInfoRequest {
+  optional string ipv4 = 1;
+  optional string ipv6 = 2;
+  string os = 3;
+  string hostname = 4;
+}
+
+message HostInfoResponse {
+  string message = 1;
```

Comment:
I'd like to suggest to use an integer error code to represent the response status,  string typed message is an optional.